### PR TITLE
dialects: Add tests for stencil.IndexAttr

### PR DESCRIPTION
**Pull request by meshtag**
This PR intends to add tests for `stencil.IndexAttr`. It should eventually help in bringing this attribute out of the experimental directory along with `stenil.FieldType`. Will do that in a separate commit.

### CODE REVIEW

File: tests/dialects/test_stencil.py [hunk 1]
```diff
@@ -184,3 +191,170 @@ def test_stencil_apply_no_results():
...
+        (
+            [
+                IntegerAttr[IntegerType](1, 64),
+                IntegerAttr[IntegerType](2, 64),
+                IntegerAttr[IntegerType](3, 64),
+            ]
+        ),
+    ),
+)
+def test_create_index_attr_from_int_list(indices: List[int]):
```
Comment (superlopuh) [inline tests/dialects/test_stencil.py:211]:
```suggestion
def test_create_index_attr_from_int_list(indices: list[int]):
```

File: tests/dialects/test_stencil.py [hunk 2]
```diff
@@ -184,3 +191,170 @@ def test_stencil_apply_no_results():
...
+    with pytest.raises(VerifyException) as exc_info:
+        IndexAttr.get(*[1] * 4)
+    assert exc_info.value.args[0] == "Expected 1 to 3 indexes for stencil.index, got 4."
+
+
+@pytest.mark.parametrize(
+    "indices1, indices2",
+    (([1], [4]), ([1, 2], [4, 5]), ([1, 2, 3], [5, 6, 7])),
+)
+def test_index_attr_size_from_bounds(indices1: List[int], indices2: List[int]):
```
Comment (superlopuh) [inline tests/dialects/test_stencil.py:239]:
```suggestion
def test_index_attr_size_from_bounds(indices1: list[int], indices2: list[int]):
```

Comment (superlopuh) [reply tests/dialects/test_stencil.py:239]: And below

**Review by superlopuh — CHANGES_REQUESTED** [review]
could you please migrate to `list` from `List`?

**Review by math-fehr — APPROVED** [review]
Nice!

**Review by tobiasgrosser — APPROVED** [review]
Nice.

**Review by superlopuh — APPROVED** [review]

**Comment by superlopuh** [comment]
ty

**Pull request merged**